include/swift/AST/Decl.h

```diff
@@ -6768,6 +6787,18 @@ class ConstructorDecl : public AbstractFunctionDecl {
   /// \endcode
   bool isObjCZeroParameterWithLongSelector() const;
 
+  /// Checks if the initializer is a distributed actor's 'local' initializer:
+  /// ```
+  /// init(transport: ActorTransport)
+  /// ```
+  bool isDistributedActorLocalInit() const;
+
+  /// Checks if the initializer is a distributed actor's 'resolve' initializer:
+  /// ```
+  /// init(resolve address: ActorAddress, using transport: ActorTransport)
+  /// ```
+  bool isDistributedActorResolveInit() const;
```

Comment:
Resolve means "please ask the `transport` to find an actor for this `address`, if if found one, _be_ that instance, otherwise, create the proxy"

I have some wording on this in the proposal, it boils down to:

```
protocol DistributedActor { 
  @derived 
  init(resolve address: ActorAddress, using transport: ActorTransport) throws { 
  	// simplified pseudo-code
    try switch transport.resolve(address, as: Self.self) { 
      case .resolved(let instance): self = instance
      case .makeProxy:              self = <runtime call to create proxy instance>
    }
  }
}
```

Reply:
In other words:

- distributed actor -- maybe remote, maybe local
- resolve init, decide if remote (proxy) or local instance
- remote init - "create proxy"
- local init - plain old initializer